std/socket.d

```diff
@@ -2750,6 +2750,14 @@ public:
         return sock;
     }
 
+    /// Return and relinquish ownership of the underlying socket handle.
```

Comment:
Probably, it won't work: until we release socket we own it, and this means that it will be closed unexpectively in case of exception, and it fails all socket operations 

It is need a private flag to indicate that the socket also used somewhere outside and not needed to close it

Reply:
I think this should be safe:

```d
auto socket = new Socket(fd, AddressFamily.INET);
scope(exit) socket.release();

...
```

It's guaranteed that `socket.release` will be called before `socket.~this`, which would call `close`.

Reply:
Oh, yes - this can be handled in user-space code by socket wrapping. so, release() suits for this purpose